pandas/io/formats/style_render.py

```diff
@@ -2098,35 +2105,54 @@ def _translate(self, styler: StylerRenderer, d: dict):
         if self.tt_data.empty:
             return d
 
-        name = self.class_name
         mask = (self.tt_data.isna()) | (self.tt_data.eq(""))  # empty string = no ttip
-        self.table_styles = [
-            style
-            for sublist in [
-                self._pseudo_css(styler.uuid, name, i, j, str(self.tt_data.iloc[i, j]))
-                for i in range(len(self.tt_data.index))
-                for j in range(len(self.tt_data.columns))
-                if not (
-                    mask.iloc[i, j]
-                    or i in styler.hidden_rows
-                    or j in styler.hidden_columns
-                )
+        if not self.as_title_attribute:
+            name = self.class_name
+            self.table_styles = [
+                style
+                for sublist in [
+                    self._pseudo_css(
+                        styler.uuid, name, i, j, str(self.tt_data.iloc[i, j])
+                    )
+                    for i in range(len(self.tt_data.index))
+                    for j in range(len(self.tt_data.columns))
+                    if not (
+                        mask.iloc[i, j]
+                        or i in styler.hidden_rows
+                        or j in styler.hidden_columns
+                    )
+                ]
+                for style in sublist
             ]
-            for style in sublist
-        ]
 
-        if self.table_styles:
-            # add span class to every cell only if at least 1 non-empty tooltip
+            if self.table_styles:
+                # add span class to every cell only if at least 1 non-empty tooltip
+                for row in d["body"]:
+                    for item in row:
+                        if item["type"] == "td":
+                            item["display_value"] = (
+                                str(item["display_value"])
+                                + f'<span class="{self.class_name}"></span>'
+                            )
+                d["table_styles"].extend(self._class_styles)
+                d["table_styles"].extend(self.table_styles)
+        else:
+            id_regex = re.compile(r"row(\d+)_col(\d+)")
             for row in d["body"]:
                 for item in row:
                     if item["type"] == "td":
-                        item["display_value"] = (
-                            str(item["display_value"])
-                            + f'<span class="{self.class_name}"></span>'
-                        )
-            d["table_styles"].extend(self._class_styles)
-            d["table_styles"].extend(self.table_styles)
-
+                        td_id = item["id"]
+                        match = id_regex.match(td_id)
+                        if match is not None:
+                            i, j = (int(x) for x in match.groups())
```

Comment:
isnt it easier to find `i,j` by using the loop structure as proposed above in the previous tooltip? You have introduced a regex here to do it but now the code is more complicated by having two methods to achieve the same loop? What advantage does this have?

Reply:
I too feel shakey on the regex, which is why I asked the question about its assumptions.

As far as the loops, to my understanding, you cannot style the title attribute of a `<td>` in any way shape or form. So we do not want to generate css nor do we want to wrap the td in a span.

If we want the tooltips as title attribute then, `self.table_styles` will naturally be empty, since it cannot be used. It would be wasted computation. There is nothing to apply styles to. If `self.table_styles` is empty then the next loop will not execute. Additionally if i were to work around that in some way, then in that loop, I'd be doing a check on every td to know whether to add the wrapping span or the title attribute. Which is not performant. 

```python
for row in d["body"]:
     for item in row:
         if item["type"] == "td":
             if self._as_title_attribute: # this seems very hot
                  <here I add the title string to the attributes>
             else:
                  item["display_value"] = (
                   str(item["display_value"])
                    + f'<span class="{self.class_name}"></span>'
                    )
```

Lastly, the loop used for `self.table_styles` isn't particularly helpful here. I need i,j at the time of accessing the row td elements, so I can pull it from `self.tt_data` and write to `item['attribute']`.  So I would need a second loop I believe.

After all that I just felt it was simpler and safer to branch immediately, the two methods are extremely different and share very little.

If you feel very strongly on this, I try to consolidate it, so we do not have two loops like below.

```
for row in d["body"]:
     for item in row:
         if item["type"] == "td":
```

Reply:
You misunderstand me. The branch is fine the question regards the detmination of the variables i and j. You have used a flakey regex approach. The above branch has used a loop. Are the variables i,j the same in both branches? If so what is the point in two different approaches. If not what is the difference? perhaps there is a better way to do this than use regex

Reply:
Ah, indeed I did greatly misunderstand you. 

After further review, I see what you are saying. I believe I found the solution by using `i` to index into the `list` in `d['body']` to select the row and then get the correct item by using `j` + number of levels in the index as the offset to the `td`

Reply:
I believe my offset has short coming, and I need to check if the styler has its index hidden and/or if specific levels are hidden to get the correct offset.

Reply:
I wrote an extra test to verify this last part. I did not realize that Styler rendered html/css for everything and just sets visibility to False. I realized that after writing the test. So, I could go back and delete it but other than that I think I am done, unless you have final comments on anything.